R/dbi-table.R

```diff
@@ -152,21 +117,65 @@ setMethod("dbWriteTable", c("OdbcConnection", "SQL", "data.frame"),
 
 #' @rdname DBI-tables
 #' @inheritParams DBI::dbAppendTable
+#' @inheritParams DBI::dbWriteTable
 #' @export
 setMethod("dbAppendTable", "OdbcConnection",
-  function(conn, name, value, ..., row.names = NULL) {
+  function(conn, name, value,
+           batch_rows = getOption("odbc.batch_rows", NA),
+           ..., row.names = NULL) {
     stopifnot(is.null(row.names))
-    stopifnot(dbExistsTable(conn, name))
-    dbWriteTable(conn, name, value, ..., row.names = row.names, append = TRUE)
+
+    fieldDetails <- tryCatch({
+      details <- odbcConnectionColumns(conn, name, exact = TRUE)
+      details$param_index <- match(details$name, colnames(value))
+      details[!is.na(details$param_index) & !is.na(details$data_type), ]
+    },
+    error = function(e) {
+      return(NULL)
+    })
+
+    if (nrow(value) > 0) {
+      name <- dbQuoteIdentifier(conn, name)
+      fields <- dbQuoteIdentifier(conn, colnames(value))
+      nparam <- length(fields)
+      params <- rep("?", nparam)
+
+      sql <- paste0(
+        "INSERT INTO ", name, " (", paste0(fields, collapse = ", "), ")\n",
+        "VALUES (", paste0(params, collapse = ", "), ")"
+      )
+      rs <- OdbcResult(conn, sql)
+
+      if (!is.null(fieldDetails) && nrow(fieldDetails) == nparam) {
+        result_describe_parameters(rs@ptr, fieldDetails)
+      }
+
+      values <- sqlData(conn, row.names = row.names, value[, , drop = FALSE])
+      if (is.na(batch_rows)) {
+        batch_rows <- NROW(value)
+        if (batch_rows == 0) {
+          batch_rows <- 1
+        }
```

Comment:
Not sure how this condition can ever be `TRUE`, with the check in line 137 in this PR.